Inconsistent TypeScript definitions in the module @keycloak/keycloak-admin-client while compiling

### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/client/node

### Describe the bug

After updating to @keycloak/keycloak-admin-client/23.0.0
i get compilation errors due to inconsistent type definitions in the admin client module.


### Version

23.0.0

### Expected behavior

compilation has no TypeScript errors, TS definitions are correct.

### Actual behavior


```
❯ npm run build

> test-service@0.1.0 build
> tsc

src/test/node_modules/@keycloak/keycloak-admin-client/lib/resources/agent.d.ts:18:15 - error TS2304: Cannot find name 'HeadersInit'.

18     headers?: HeadersInit;
                 ~~~~~~~~~~~

src/test/node_modules/@keycloak/keycloak-admin-client/lib/utils/fetchWithError.d.ts:10:47 - error TS2552: Cannot find name 'RequestInfo'. Did you mean 'RequestInit'?

10 export declare function fetchWithError(input: RequestInfo | URL, init?: RequestInit): Promise<Response>;
                                                 ~~~~~~~~~~~
``` 

### How to Reproduce?

create any empty project and include @keycloak/keycloak-admin-client as dependency.
npm install


Open in Visual Studio Code the project, and open both files
node_modules/@keycloak/keycloak-admin-client/lib/resources/agent.d.ts:18:15 
node_modules/@keycloak/keycloak-admin-client/lib/utils/fetchWithError.d.ts:10:47

See errors in the editor


### Anything else?

This became visible after I migrated from the previous deprecated package "keycloak-admin": "^1.14.18".

--- a/js/libs/keycloak-admin-client/src/resources/agent.ts
+++ b/js/libs/keycloak-admin-client/src/resources/agent.ts
@@ -36,7 +36,7 @@ export interface RequestArgs {
    * Keys to be ignored, meaning that they will not be filtered out of the request payload even if they are a part of `urlParamKeys` or `queryParamKeys`,
    */
   ignoredKeys?: string[];
-  headers?: HeadersInit;
+  headers?: [string, string][] | Record<string, string> | Headers;
 }
 
 const pick = (value: Record<string, unknown>, keys: string[]) =>
@@ -197,7 +197,7 @@ export class Agent {
     catchNotFound: boolean;
     payloadKey?: string;
     returnResourceIdInLocationHeader?: { field: string };
-    headers?: HeadersInit;
+    headers?: [string, string][] | Record<string, string> | Headers;
   }) {
     const newPath = urlJoin(this.#basePath, path);
 

--- a/js/libs/keycloak-admin-client/src/utils/auth.ts
+++ b/js/libs/keycloak-admin-client/src/utils/auth.ts
@@ -116,6 +116,6 @@ export const getToken = async (settings: Settings): Promise<TokenResponse> => {
     body: payload,
   });
 
-  const data: TokenResponseRaw = await response.json();
+  const data = (await response.json()) as TokenResponseRaw;
   return camelize(data);
 };

--- a/js/libs/keycloak-admin-client/src/utils/fetchWithError.ts
+++ b/js/libs/keycloak-admin-client/src/utils/fetchWithError.ts
@@ -14,7 +14,7 @@ export class NetworkError extends Error {
 }
 
 export async function fetchWithError(
-  input: RequestInfo | URL,
+  input: Request | string | URL,
   init?: RequestInit,
 ) {
   const response = await fetch(input, init);

--- a/js/libs/keycloak-admin-client/tsconfig.json
+++ b/js/libs/keycloak-admin-client/tsconfig.json
@@ -2,6 +2,7 @@
   "extends": "../../tsconfig.json",
   "include": ["src"],
   "compilerOptions": {
+    "lib": ["ESNext"],
     "module": "Node16",
     "moduleResolution": "Node16",
     "outDir": "lib",

